File: app/src/main/java/com/wolkowiczmateusz/android/cleanmvpretrofitrxdaggerboilerplate/presentation/screens/login/LoginContract.java
package com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.presentation.screens.login;

import com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.presentation.base.BaseContractMvpPresenter;
import com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.presentation.base.BaseContractMvpView;


public interface LoginContract {

    interface View extends BaseContractMvpView {

        void disableLoginButton(boolean state);
        void showErrors(String errorEmail, String errorPassword);
        void showRegister();
        void Login();
    }

    interface Presenter<V extends View> extends BaseContractMvpPresenter<V> {

        void LoginClick(String email, String password);
        void RegisterClick();
    }
}

File: app/src/main/java/com/wolkowiczmateusz/android/cleanmvpretrofitrxdaggerboilerplate/presentation/screens/login/LoginContract.kt
package com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.presentation.screens.login

import com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.presentation.base.BaseContractMvpPresenter
import com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.presentation.base.BaseContractMvpView

interface LoginContract {

    interface View : BaseContractMvpView {

        fun disableLoginButton(state: Boolean)
        fun showErrors(errorEmail: String?, errorPassword: String?)
        fun showRegister()
        fun login()
    }

    interface Presenter<V : LoginContract.View> : BaseContractMvpPresenter<V> {

        fun loginClick(email: String, password: String)
        fun registerClick()
    }
}
